### ISSUE TITLE
Allow routines to access the folder where data will be saved

### ISSUE THREAD

**Issue by aorgazf** [qiboteam/qibocal#533]
Hi,
While developing a routine to perform the T1_T2_vs_Temperature experiment, would have liked to be able to save data to the same folder where the report is saved. The routine performs measurements for long periods and extracts quite a lot of data. To minimise the risk of loosing data, the routine saves the data corresponding to each measurement (t1, ramsey_t2, qubit spectroscopy) after they are taken.
The problem is that the folder where the report is computed is currently calculated only after the routine is finished, so there is no way to access it. If possible, it would be nice to have it calculated beforehand and have it accessible to routines.

Thanks.
